### ISSUE TITLE
estimate_fit

### ISSUE THREAD

**Issue by strengejacke** [easystats/see#38]
@DominiqueMakowski `estimate_fit()` was renamed to `estimate_response()`, if I'm not mistaken. However, the plot now looks a bit different (I guess this is intended).

Could you please check https://github.com/easystats/see if plots are OK?

**Comment by DominiqueMakowski** [comment]
![image](https://user-images.githubusercontent.com/8875533/61944889-52947f00-afd1-11e9-9594-b9ba296083be.png)

maaan that plot is sick 😍😅

but no it is not the intended one indeed. I'll correct the README

**Comment by strengejacke** [comment]
ehm, you probably need to correct the plot-function directly. ;-)

But yes, the plot looks crazy, though I really like the idea of having CIs for the uncerntainty in the draws (though this was not the case in the previous plots, if unintended, cool! :-D)

**Comment by DominiqueMakowski** [comment]
> ehm, you probably need to correct the plot-function directly. ;-)

right, lemme look into it

**Comment by strengejacke** [comment]
This is not urgend, since estimate isn't on CRAN yet, anyway.

**Comment by DominiqueMakowski** [comment]
it was just about changing estimate_response to estimate_link

**Comment by strengejacke** [comment]
I just saw that it was almost completely nonsense what I wrote... It's probably indeed something to be fixed in the readme!

**Comment by strengejacke** [comment]
We don't have a plot-method for `estimate_response()` or `estimate_link()` yet...

**Comment by DominiqueMakowski** [comment]
> We don't have a plot-method for estimate_response() or estimate_link() yet...

I know, but my guess is this would be tricky; we would have to detect from the data the appropriate aesthetics (if interactions, draws, ci etc)

I'd say for now we can live with partial support for estimate_contrasts, which is just there as an initial implementation of lighthouse plots. Maybe we can think about improving the support for `estimate`  once we are at the CRAN stage

**Comment by strengejacke** [comment]
> ![image](https://user-images.githubusercontent.com/8875533/61944889-52947f00-afd1-11e9-9594-b9ba296083be.png)
> 
> maaan that plot is sick 
> 
> but no it is not the intended one indeed. I'll correct the README

I really like this plot, we need to implement a function that produces this kind of plot! 😀

**Comment by strengejacke** [comment]
@DominiqueMakowski what about adding some more plot functions for the modelbased package? I think especially modelbased is a package that would really benefit from plot functions.

**Comment by DominiqueMakowski** [comment]
I agree I agree, and also making them more robust to different formulas and parameters. Maybe inspiration could be drawn for estimate_link and estimate_response from ggeffects, I wonder how you deal with all the possibilities.

And for estimate_means() and estimate_response(), i.e., in cases where x is a factor, I'd love to improve and reinforce lighthouse plots #12 but last time I hit the limits of my abilities. Maybe now I could give it a fresh look, but first let me finish with report 😅

**Comment by strengejacke** [comment]
There are quite some examples for plots in the vignettes, maybe we can at least start with functions that create "simple" plots like those? Maybe over the next weeks...

**Comment by DominiqueMakowski** [comment]
Yes I agree. Though we could start with improving/making more robust the plotting `estimate_means()` because I think it could be the most straightforward to support and also because it's the one for which there is less alternative for now (whereas for estimate_link people can still use ggeffecst for easy plotting :)

Or `estimate_response()` which by default returns the predicted value for each point of the dataset. Basically I see that the difference between `estimate_link()` and `estimate_response()`, when it comes for plotting, is that estimate_link would plot the link(s) only (which is the interest) whereas estimate_response would plot the predicted relationship with also the datapoints of the original data (something [like this](https://github.com/easystats/modelbased#generate-predictions-from-your-model-to-compare-it-with-original-data)). 

We could essentially have a common function with an argument `add_datapoints`, set to FALSE by default for `estimate_link` objects and to TRUE for estimate_response.

**Comment by bwiernik** [comment]
I am really missing plotting functions for `estimate_predicted()`. Producing a plot like this seems pretty fundamental for model visualization:

``` r
library(ggplot2)
model <- lm(mpg ~ factor(cyl) * wt, data = mtcars)
ggplot(modelbased::estimate_prediction(model, data = "grid"), aes(x = wt, color = factor(cyl), fill = factor(cyl))) + 
    geom_point(aes(y = mpg), data = insight::get_data(model)) +
    geom_ribbon(aes(ymin = CI_low, ymax = CI_high), alpha = .2, size = 0) +
    geom_line(aes(y = Predicted))
```

![](https://i.imgur.com/HA9ziBF.png)

<sup>Created on 2021-05-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by bwiernik** [comment]
@DominiqueMakowski Let me know if you'd like to spitball or discuss any design considerations for this

**Comment by DominiqueMakowski** [comment]
absolutely, I'm currently thinking of a proof-of-concept for an updated approach to my long-term goal that a draft can be seen through one of the original idea for `see` that I had in mind (https://github.com/easystats/see/issues/37#issuecomment-511448606; https://github.com/easystats/see/issues/9) that I never had the chance to fully realize due to my issues with focus management.

Besides the longer-term part of this goal (to offer an ingredients&recipe based - instead of a pre-baked - approach to plotting), the shorter goal would be to expose the middle portion, between the object and the plot. 

In other words, I'm thinking of a `visualisation_recipe()` method that would take an object and return the data ready to be plotted (for now I'm trying a list of "geom" layers information), which could be easily and robustly plotted by `see` (with all the options of customisation). Will open an issue in modelbased to discuss some ideas

**Comment by bwiernik** [comment]
Feel free to move this comment over to your modelbased issue.

I'm not exactly following what you mean by "method that would take an object and return the data ready to be plotted". The `estimate_predicted()` result already contains the necessary information, no? predictor values, response value, fitted value with SE and CI?

I think a basic approach for `estimate_predicted()` would be to produce a plot with:
1. x-axis
    1. If model has one predictor, the predictor
    2. If model has multiple predictors, default to the fitted value? or maybe the first predictor in the formula?
    3. Have an `x` argument that takes the name of a variable to map to the x axis.
2. geom_line
    1. y = Fitted 
3.  geom_ribbon
    1. ymin = CI_low, ymax = CI_high 
4. points
    1. If x is numeric, `geom_point(aes(y = response))`
    2. If x is factor or ordered, `geom_jitter(aes(y = response), height = 0, width =  (length(levels(x)) - 1) / (4 * length(levels(x))))`
5. other aeshetics
    1. Arguments for `shape`, `color`, `alpha`, `linetype`, `facet`, etc. that pass these along

Then, we apply nice *see* defaults. If a user wants more customization, they can take the resulting ggplot and add more layers.

**Comment by IndrajeetPatil** [comment]
Closed in favor of https://github.com/easystats/modelbased/issues/110
